devtools/build_defaults_rst.py

```diff
@@ -0,0 +1,205 @@
+
+
+def loop_params(config, easy, expert, guru, next_level='`'):
+    """
+    Treat parameters for module.
+
+    *Important:* considers that some configuration files can have dictionaries
+    with subparameters. However, there should NOT be more than one level
+    of nesting in the configuration parameter files.
+    """
+    # sort parameters by name
+    sorted_ = sorted(
+        ((k, v) for k, v in config.items()),
+        key=lambda x: x[0],
+        )
+
+    for name, data in sorted_:
+        if isinstance(data, Mapping) and "default" not in data:
+            explevel = data["explevel"]
+
+            new_title = [name, next_level * len(name), '']
+            if explevel == 'easy':
+                easy.extend(new_title)
+                sublist = easy
+            elif explevel == 'expert':
+                expert.extend(new_title)
+                sublist = explevel
+            elif explevel == 'guru':
+                guru.extend(new_title)
+                sublist = guru
+            elif explevel == 'hidden':
+                continue
+            else:
+                emsg = f'explevel {explevel!r} is not expected'
+                raise AssertionError(emsg)
+
+            data_text = (
+                f'| *title*: {data["title"]}',
+                f'| *short description*: {data["short"]}',
+                f'| *long description*: {data["long"]}',
+                f'| *group*: {data["group"]}',
+                f'| *explevel*: {data["explevel"]}',
+                '',
+                )
+            sublist.append(os.linesep.join(data_text))
+
+            for name2, param2 in data.items():
+                if isinstance(param2, Mapping):
+                    if next_level == '`':
+                        nlevel = '~'
+                    elif next_level == '~':
+                        nlevel = '>'
+
+
+                    text = do_text(name2, param2, level=nlevel)
+                    sublist.append(text)
+
+        elif isinstance(data, Mapping):
+            explevel = data["explevel"]
+
+            text = do_text(name, data, level=next_level)
+
+            if explevel == 'easy':
+                easy.append(text)
+            elif explevel == 'expert':
+                expert.append(text)
+            elif explevel == 'guru':
+                guru.append(text)
+            elif explevel == 'hidden':
+                continue
+            else:
+                emsg = f'explevel {explevel!r} is not expected'
+                raise AssertionError(emsg)
+
+    easy.append('')
+    expert.append('')
+    guru.append('')
+
+    return easy, expert, guru
+
+
+def build_rst(module_params, level='-'):
+    """Build .rst text."""
+    easy = ['Easy', level * 4, '']
+    expert = ["Expert", level * 6, '']
+    guru = ['Guru', level * 4, '']
+
+    # very ugly implementation, but there are no more cases than this.
+    if level == '-':
+        nlevel = '`'
+    elif level == '`':
+        nlevel = '~'
```

Comment:
I guess this requires some comment, as I cannot understand what's happening here..